app/src/Home.js

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+import AddContact from "./Components/AddContact";
+import ContactDetail from "./Components/ContactDetail";
+import ContactsList from "./Components/ContactsList";
+import * as apiClient from "./apiClient";
+
+const Home = () => {
+  const [contacts, setContacts] = React.useState([]);
+  const [selectedContact, setSelectedContact] = React.useState("");
+  const [contactDetails, setContactDetails] = React.useState({});
+
+  const loadContacts = async () => setContacts(await apiClient.getContacts());
+
+  const addContact = async (contact) => {
+    await apiClient.addContact(contact);
+    loadContacts();
+  };
+
+  const editContact = async (contact, id) => {
+    await apiClient.editContact(contact, id);
+    loadContacts();
+  };
+
+  const deleteContact = (id) => {
+    apiClient.deleteContact(id).then(loadContacts);
+  };
+
+  const loadContact = React.useCallback(
+    () => apiClient.getContact(selectedContact).then(setContactDetails),
+    [selectedContact],
+  );
+
+  React.useEffect(() => {
+    selectedContact !== undefined && loadContact();
```

Comment:
[fix] There's a subtle bug here that causes you to load your entire contact list twice on startup. See if you can figure out why.